### ISSUE TITLE
Services_Twilio_RestException: The From phone number +85258085120 is not a valid

### ISSUE THREAD

**Issue by ezekel** [twilio/twilio-php#243]
can I ask help what does this error means
![twilio](https://cloud.githubusercontent.com/assets/6830057/8300035/f2e060aa-19b2-11e5-93db-ec0014bbd28c.PNG)

here is my set up

```

  require ('Services/Twilio.php');


$sid = "xxxxxxxxxxxxxxxxxxx"; // Your Account SID from www.twilio.com/user/account
$token = "xxxxxxxxxxxxxxxxx"; // Your Auth Token from www.twilio.com/user/account

   $http = new Services_Twilio_TinyHttp(
    'https://api.twilio.com',
    array('curlopts' => array(
        CURLOPT_SSL_VERIFYPEER => true,
        CURLOPT_SSL_VERIFYHOST => 2,

    )));

$client = new Services_Twilio($sid, $token, "2010-04-01", $http);

$message = $client->account->messages->sendMessage(
    '85258085120', // From a valid Twilio number
    'my-mobile-no', // Text this number
    "Hello Test!"
);

print $message->sid;


```

Thank you in advance

**Comment by peardox** [comment]
It's a HK landline

https://www.twilio.com/lookup

**Comment by philnash** [comment]
This just came up in my email thanks to @peardox's comment. As they say, the number `+85258085120` is an HK landline number owned by Voxbone according to the Twilio lookup.

@ezekel Do you have a Twilio number available in your account dashboard? Can you double check it's the number you think it is? Let me know if you are still having problems!

**Comment by peardox** [comment]
People frequently enter landline numbers

This may be a foreign concept to the younger guys out there but I've got 3904 'Unknown Error' results (about 5% fail) from a recent run. 

I suspect these are landlines but I'm not gonna run them all thru lookup as the client will incur a $20 bill for the service (unless he asks me to)

**Comment by philnash** [comment]
@peardox In the case above for @ezekel, they are using that HK number as their "From" number in an API request. They will need to use their own Twilio numbers in order to send SMS messages.

**Comment by ezekel** [comment]
@philnash @peardox ,

Here is my screenshot, I'm still having problem on this I cannot try sending message.

Thank you in advance.

![twilio](https://cloud.githubusercontent.com/assets/6830057/9373686/afa4e7bc-471f-11e5-9bc6-5fc75af95a54.PNG)
![twilio error](https://cloud.githubusercontent.com/assets/6830057/9373690/be5d286e-471f-11e5-9568-72292b6e6593.PNG)

**Comment by peardox** [comment]
To my mind this indicates you're using a self-signed certificate for a callback

i.e. You're using https for your end without having purchased a real one (and Twilio-php is checking you're real)

Twilio will work fine without https (not recommended for production) so for testing just use http

If you deploy to something like Heroku you should bypass this particular issue

This is only supposition but seems likely - try it on http first and if you still get the issue I'll modify the test script I write for #253 to do a callback as well both direct and with twilio-php

**Comment by ezekel** [comment]
@peardox ,

here is the script I use I don't what you mean using the https

```
require ('Services/Twilio.php');


$sid = "xxxxxxxxxxxxxxx"; // Your Account SID from www.twilio.com/user/account
$token = "xxxxxxxxxxxxxxxxxxxxxxx"; // Your Auth Token from www.twilio.com/user/account


$http = new Services_Twilio_TinyHttp(
    'https://api.twilio.com',
    array('curlopts' => array(
        CURLOPT_SSL_VERIFYPEER => true,
        CURLOPT_SSL_VERIFYHOST => 2,

    )));

$client = new Services_Twilio($sid, $token);

$message = $client->account->messages->sendMessage(
    '85258085120', // From a valid Twilio number
    'mynumber', // Text this number
    "Hello Test!"
);


print $message->sid;

```

Thank you in advance

**Comment by philnash** [comment]
Hey @ezekel,

A bunch of other people have had this problem and there's some fixes mentioned in #203 which I suggest you have a look through and try out. Let me know if anything there helps.
